WordPress/Classes/Utility/ContextManager.swift

```diff
@@ -69,31 +79,36 @@ public class ContextManager: NSObject, CoreDataStack, CoreDataStackSwift {
         context.performAndWait {
             block(context)
 
-            self.save(context, .synchronously)
+            self.save(context, .alreadyInContextQueue)
         }
     }
 
     @objc(performAndSaveUsingBlock:completion:onQueue:)
-    public func performAndSave(_ block: @escaping (NSManagedObjectContext) -> Void, completion: @escaping () -> Void, on queue: DispatchQueue) {
+    public func performAndSave(_ block: @escaping (NSManagedObjectContext) -> Void, completion: (() -> Void)?, on queue: DispatchQueue) {
```

Comment:
Why has the completion changed from `@escaping` to optional?

Reply:
The "save context" functions declare their `completion` argument as Optional, I thought it's best to change this argument here to keep them consistent. And, an Optional closure is automatically escaped (I forgot why, some not-so-straightforward Swift implementation details I guess...)

Reply:
Cool. +1 for consistency.

I did not know about that Swift behavior. I'd say nice, but it's also a bit opaque 🤔